src/GafferImage/Resample.cpp

```diff
@@ -180,13 +186,28 @@ Box2i inputRegion( const V2i &tileOrigin, unsigned passes, const V2f &ratio, con
 
 // Given a filter name, the current scaling ratio of input size / output size, and the desired filter scale in
 // output space, return a filter and the correct filter scale in input space
-const OIIO::Filter2D *filterAndScale( const std::string &name, V2f ratio, V2f &inputFilterScale )
+const OIIO::Filter2D *filterAndScale( const std::string &name, V2f ratio, bool deep, Resample::DeepMode deepMode, V2f &inputFilterScale )
 {
 	ratio.x = fabs( ratio.x );
 	ratio.y = fabs( ratio.y );
 
+	if( deep && deepMode == Resample::DeepMode::ErrorIfSlow && name != g_nearestString )
+	{
+		throw IECore::Exception(
+			"Accurately filtering deep images is very slow and produces results that are impractically large "
+			"to save to disk. We recommend switching to the filter \"nearest\", using a more efficient "
+			"approach we haven't implemented yet, or if you know what you're doing, and want a fully "
+			"filtered result, set \"deepMode\" to \"Accurate\"."
```

Comment:
Last we spoke, I thought you had floated the idea of an "all-in-one-merge/flatten" as a separate node network we would include? That would open the door to doing some optimization. Even without any external visibility on the code inside Resample, we could identify whole tiles where only one of the images has content, and skip the deep resize. It also would take very little code to add a mask input to Resample and do that bypass per pixel. There are pretty common production cases where this would be a 5X speedup.

Reply:
> Last we spoke, I thought you had floated the idea of an "all-in-one-merge/flatten" as a separate node network we would include? 

That does ring a bell, and seems reasonable as long as it doesn't complicate the internals of some already quite complicated nodes. I still don't think it makes much sense to talk about this in the docs though.